### PR TITLE
💄 Improve Style of Color Selection and Colorpicker

### LINKED ISSUE #442: ColorPicker: der eigentliche Picker wirkt noch deplaziert
**Beschreibe das Problem:**

Bei der Auswahl der genauen Farbe wird ein komplexes Kontrollelement eingeblendet, dass in seiner Präsentation noch sehr fremd wirkt.

Hier sollten wir evaluieren, ob diese Komponente wirklich der "best fit" für unser UI ist und inwieweit sie optisch angepasst werden kann, um mehr wie eine Farbauswahl in Photoshop, Word o.ä. zu wirken (siehe unten).

image

image

### PR DESCRIPTION
**Pull request by SteffenKn**
## What did you change?

This PR improves the style of the color selection and the colorpicker to make its look adapt to that of the BPMN-Studio.

Also the glyphicons got replaced by font awesome icons.

Closes #442

## How can others test the changes?

- Open up the BPMN-Studio
- Have a look at the color selection and the colorpicker

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [ ] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/modules/diagram-tools-right/diagram-tools-right.scss [hunk 1]
```diff
@@ -32,6 +32,23 @@
   cursor: default;
 }
 
+.colorpicker--fill {
```
Comment (Vyperus) [inline src/modules/diagram-tools-right/diagram-tools-right.scss]: The class "colorpicker--fill" suggests that something gets filled. But you are adding a border-top here.

Please find a better name.

File: src/modules/diagram-tools-right/diagram-tools-right.scss [hunk 2]
```diff
@@ -32,6 +32,23 @@
...
+  border-top: 1px solid #ccc;
+}
+
+.colorpicker--border {
```
Comment (Vyperus) [inline src/modules/diagram-tools-right/diagram-tools-right.scss]: See above.

File: src/modules/diagram-tools-right/diagram-tools-right.scss [hunk 3]
```diff
@@ -32,6 +32,23 @@
...
+  border-bottom: 1px solid #ccc;
+}
+
+.colorpicker__eyedropper--fill {
```
Comment (Vyperus) [inline src/modules/diagram-tools-right/diagram-tools-right.scss]: See above.

File: src/modules/diagram-tools-right/diagram-tools-right.scss [hunk 4]
```diff
@@ -32,6 +32,23 @@
...
+  margin-left: 39px;
+  margin-right: -14px;
+}
+.colorpicker__eyedropper--border {
```
Comment (Vyperus) [inline src/modules/diagram-tools-right/diagram-tools-right.scss]: See above.

File: src/modules/processdef-detail/processdef-detail.scss [hunk 5]
```diff
@@ -51,3 +51,14 @@
 .none {
   color: transparent;
 }
+
+.colorpicker-container {
+  border: solid 1px #ccc !important;
+  background: white !important;
+}
+
+.sp-dd {
```
Comment (Vyperus) [inline src/modules/processdef-detail/processdef-detail.scss]: Please add a comment for classes like this.

Answer the question: "Why is this here?"

File: src/modules/diagram-tools-right/diagram-tools-right.scss (hunk 1 above)
Comment (SteffenKn) [reply src/modules/diagram-tools-right/diagram-tools-right.scss]: Like "colorpicker--fill-color"? Or do you have a better idea?

File: src/modules/diagram-tools-right/diagram-tools-right.ts [hunk 6]
```diff
@@ -146,42 +149,69 @@ export class DiagramToolsRight {
...
     const fillMoveSetting: spectrum.Options = {
       move: (fillColor: spectrum.tinycolorInstance): void => {
         this._updateFillColor(fillColor);
       },
     };
 
-    const colorPickerFillSettings: IColorPickerSettings = Object.assign({}, environment.colorPickerSettings, fillMoveSetting);
+    const fillLocalStorageKey: spectrum.Options = { localStorageKey: 'fillColors' };
+
+    const fillDefaultColors: Array<string> = ['#FFCDD2', '#BBDEFB', '#C8E6C9', '#E1BEE7', '#FFE0B2'];
```
Comment (cmg-dev) [inline src/modules/diagram-tools-right/diagram-tools-right.ts]: Jetzt würde ich das in in constanten verlagern! 

man braucht das an zwei verschiedenen stellen...

File: src/modules/diagram-tools-right/diagram-tools-right.scss [hunk 7]
```diff
@@ -32,6 +32,24 @@
   cursor: default;
 }
 
+.fill-colorpicker {
```
Comment (cmg-dev) [inline src/modules/diagram-tools-right/diagram-tools-right.scss]: Please add a comment to improve the purpose here.

**Comment by robpal94** [comment]
Not sure if this counts as a bug, but when I try to change the color on a sequence flow, a new `marker` Element gets appended to the `dev` Element. This results in a pretty laggy behavior, since I keep doing this (the html that the browser has to render grows and grows which leads to this laggy behavior). 

jun-19-2018 11-09-06

**Comment by SteffenKn** [comment]
@robpal94 This is a general issue with the color selection.
We should fix this bug, but not in this PR. It would be nice if you could open an issue for that.

**Comment by robpal94** [comment]
https://github.com/process-engine/bpmn-studio/issues/504#issuecomment-398375286

File: src/modules/diagram-tools-right/diagram-tools-right.ts [hunk 8]
```diff
@@ -84,23 +115,23 @@ export class DiagramToolsRight {
...
     const selectedElements: Array<IShape> = this._getSelectedElements();
 
-    const isNoValidElement: boolean = selectedElements.length < 1
+    const elementIsNotValid: boolean = selectedElements.length < 1
                                     || selectedElements.length === 1
                                     && selectedElements[0].$type === 'bpmn:Collaboration';
 
-    if (isNoValidElement) {
+    if (elementIsNotValid) {
       this._notificationService.showNotification(NotificationType.ERROR, 'Error while changing the color: No valid element was selected.');
```
Comment (cmg-dev) [inline src/modules/diagram-tools-right/diagram-tools-right.ts]: This is acutally not an error.

This is a Warning-- probably a notification as suggestion-- please update like this:

`this._notificationService.showNotification(NotificationType.INFO, 'Unable to apply color. Please select an element and use the color picker again.');`

Comment (cmg-dev) [reply src/modules/diagram-tools-right/diagram-tools-right.ts]: @Vyperus You should always have the user in mind, when approving PRs.

This will satisfy @rrrene-- muchos grande.

Comment (SteffenKn) [reply src/modules/diagram-tools-right/diagram-tools-right.ts]: Done

File: src/modules/diagram-tools-right/diagram-tools-right.ts [hunk 9]
```diff
@@ -84,23 +115,24 @@ export class DiagramToolsRight {
...
-    const isNoValidElement: boolean = selectedElements.length < 1
+    const elementIsNotValid: boolean = selectedElements.length < 1
                                     || selectedElements.length === 1
                                     && selectedElements[0].$type === 'bpmn:Collaboration';
 
-    if (isNoValidElement) {
-      this._notificationService.showNotification(NotificationType.ERROR, 'Error while changing the color: No valid element was selected.');
+    if (elementIsNotValid) {
+      this._notificationService.showNotification(NotificationType.INFO, `Unable to apply color.
+      Please select an element and use the color picker again.`);
```
Comment (LeoTT) [inline src/modules/diagram-tools-right/diagram-tools-right.ts]: Templatestrings can contain and carry linebreaks and spaces. The notification would therefore have a weird linebreak followed by 4 spaces. 
(May not actually show the linebreak, not sure about linebreaks and injected html, but the spaces should be there)
```ts
console.log(`test
             test`).

// results in:
test
             test
```
If you want to please the linter, put the string into an extra variable.
If you ever have to break strings into multiple lines you can do this:
```ts
this._notificationService.showNotification(NotificationType.INFO, 'Unable to apply color.'
      + 'Please select an element and use the color picker again.');
```
